orderapi/src/test/java/fi/hel/verkkokauppa/order/service/order/OrderServiceTest.java

```diff
@@ -645,4 +644,26 @@ void createOrderWithMerchantId() throws JsonProcessingException {
         orderRepository.save(order);
     }
 
+    @Test
+    @RunIfProfile(profile = "local")
+    public void CreateOrderWithLastValidPurchaseDateTime() {
+        String firstMerchantIdFromNamespace = getFirstMerchantIdFromNamespace("venepaikat");
+        OrderAggregateDto createOrderResponse = createNewOrderToDatabase(1, firstMerchantIdFromNamespace).getBody();
+        assert createOrderResponse != null;
+        Order order = orderRepository.findById(createOrderResponse.getOrder().getOrderId()).get();
+        LocalDateTime localDateTime = LocalDateTime.now();
+        order.setLastValidPurchaseDateTime(localDateTime);
+        orderRepository.save(order);
+        Order fetchedOrder = orderRepository.findById(order.getOrderId()).get();
+
+        /* Creating localDateTime with now() uses 6 digit precision with nanoseconds by default.
+           This is automatically rounded on save on db's side to 3 digit precision which causes assertion
+           error if values are asserted as is. Creating a formatter just for this felt excessive, thus
+           nanoseconds are rounded to 0 for "good-enough" assertion. */
```

Comment:
This is good assertion and new thing to me, can make testing easier in some places where I had to create formatters.